Subject: [PATCH] git-remote-mediawiki: support empty commits

It's possible for the remote to return a revision that has user
data and possibly a message, but no actual content changes.

Treat this as an empty commit when importing revisions to preserve
the user and comment message.

Note that this is an unusual condition not normally encountered,
but it can occur and having it cause a deletion of the page would
be incorrect.

Signed-off-by: Kyle J. McKay <mackyle@gmail.com>

---
 contrib/mw-to-git/git-remote-mediawiki.perl | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/contrib/mw-to-git/git-remote-mediawiki.perl b/contrib/mw-to-git/git-remote-mediawiki.perl
index 75db85b6..a6657140 100755
--- a/contrib/mw-to-git/git-remote-mediawiki.perl
+++ b/contrib/mw-to-git/git-remote-mediawiki.perl
@@ -630,6 +630,7 @@ sub mediawiki_clean {
 # Filter applied on MediaWiki data before adding them to Git
 sub mediawiki_smudge {
 	my $string = shift;
+	return undef unless defined($string);
 	if ($string eq EMPTY_CONTENT) {
 		$string = EMPTY;
 	}
@@ -785,10 +786,12 @@ sub import_file_revision {
 	if (!$full_import && $n == 1) {
 		print {*STDOUT} "from refs/mediawiki/${remotename}/master^0\n";
 	}
-	if ($content ne DELETED_CONTENT) {
-		print {*STDOUT} 'M 644 inline ' .
-		    fe_escape_path("${title}.mw") . "\n";
-		literal_data($content);
+	if (!defined($content) || $content ne DELETED_CONTENT) {
+		if (defined($content)) {
+			print {*STDOUT} 'M 644 inline ' .
+			    fe_escape_path("${title}.mw") . "\n";
+			literal_data($content);
+		}
 		if (%mediafile) {
 			print {*STDOUT} 'M 644 inline '
 			    . fe_escape_path($mediafile{title}) . "\n";
---
